service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -134,6 +139,14 @@ private void updateDatabaseRunStatus(
                 e);
             updatedRunState = CbasRunStatus.SYSTEM_ERROR;
           }
+        } else if (updatedRunState.inErrorState()) {
+          try {
+            String message = cromwellService.getRunErrors(r);
+            Run failedRunWithError = r.withErrorMessage(message);
+            addToUpdatedRunSet(r, updatedRuns, failedRunWithError);
+          } catch (ApiException e) {
+            throw new RuntimeException(e);
```

Comment:
I don't know that a runtime exception is best here... that would make the user get a "500 system error" response without any information in it. 

Could we just...log the problem and give the user back a status without updating the database? The user gets some information, and we get another chance to look up the error message next time they call the status endpoint?

But I haven't actually traced this through, it might not be what _actually_ happens... but in general we should be super suspicious of any RuntimeExceptions that can happen in the codebase, let alone runtime exceptions that we choose to add ourselves!